src/com/company/Parse.java

```diff
@@ -0,0 +1,120 @@
+package com.company;
+
+import org.eclipse.cdt.core.dom.ast.ExpansionOverlapsBoundaryException;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+
+    private IASTTranslationUnit translationUnit; //current translation unit node
+    public HashMap<String, Contents> repo;
+
+    public Parse(String code) throws Exception {
+        repo = new HashMap<>();
+
+        translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode node = translationUnit;
+
+        saveNodes(node);
+    }
+
+    /**Accesses all the nodes of the tree*/
+    private void saveNodes(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child :children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+
+                /**Setup*/
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+                CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
+                CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+                CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+
+                if (init == null) {
+                    continue;
+                }
+
+                if (declarator == null) {
+                    continue;
+                }
+
+                if (specifier == null) {
+                    continue;
+                }
+
+                if (declaration == null) {
+                    continue;
+                }
```

Comment:
You should be able to consolidate all these into one `if` statement, by OR-ing all of them (eg, `if (init == null || declarator == null || ...) {`)